### PR TITLE
Irregular cut wrap

### PR DESCRIPTION
**Pull request by danlipsa**

### CODE REVIEW

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 I am having issues with test_vcs_patterns_hatches.py because this test duplicates one row of data (line 23) which results in duplicate cells when wrapping. Is there a way to test that a variable was instructed to create duplicate data? In that case I should not do wrapping. Thanks!

**Comment by doutriaux1** [comment]
which test? This one: https://github.com/UV-CDAT/uvcdat/blob/master/testing/vcs/test_vcs_hatches_patterns.py

**Comment by danlipsa** [comment]
All these:
baselines/vcs/test_vcs_meshfill_hatch_SH_-180_180.png
baselines/vcs/test_vcs_meshfill_hatch_SH_-180_180_non-contig.png
baselines/vcs/test_vcs_meshfill_hatch_SH_0_360.png
baselines/vcs/test_vcs_meshfill_pattern_SH_-180_180.png
baselines/vcs/test_vcs_meshfill_pattern_SH_-180_180_non-contig.png
baselines/vcs/test_vcs_meshfill_pattern_SH_0_360.png
baselines/vcs/test_vcs_meshfill_solid_SH_-180_180.png
baselines/vcs/test_vcs_meshfill_solid_SH_-180_180_non-contig.png
baselines/vcs/test_vcs_meshfill_solid_SH_0_360.png

**Comment by danlipsa** [comment]
@doutriaux1 Yes, that is the test file, the tests are the one I posted.

**Comment by danlipsa** [comment]
Baselines that go with this
https://github.com/UV-CDAT/uvcdat-testdata/pull/103

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 I think I found the solution. I wrap only for TransitentCurveGrid

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 please review. Note the 36 updated baselines.

**Comment by danlipsa** [comment]
@doutriaux1 the macs seem hanged. Could you restart them? Thanks!

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 Ping

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 All test passed, please review. I would like to merge this in.

**Comment by aashish24** [comment]
Looks good to me but it would be great if @doutriaux1 can review this.

**Comment by doutriaux1** [comment]
@danlipsa @aashish24 @williams13  do we want this in 2.4 or in 2.4.1 ?

**Comment by aashish24** [comment]
@doutriaux1 we definitely want in 2.4.1. We can start 2.4.0 with a older sha if needed.

**Comment by aashish24** [comment]
@doutriaux1 could you please review this branch ?

File: Packages/vcs/Lib/vcs2vtk.py [hunk 1]
```diff
@@ -73,34 +73,61 @@ def numpy_to_vtk_wrapper(numpyArray, deep=False, array_type=None):
...
+                attributes2 = grid2.GetCellData() if cellData else grid2.GetPointData()
             else:
                 grid2 = vtk.vtkUnstructuredGrid()
+                if (cellData):
+                    attributes2 = grid2.GetCellData()
+                    attributes = grid.GetCellData()
+                else:
+                    attributes2 = grid2.GetPointData()
+                    attributes = grid.GetPointData()
+                attributes2.SetPedigreeIds(attributes.GetPedigreeIds())
```
Comment (doutriaux1) [inline Packages/vcs/Lib/vcs2vtk.py]: @danlipsa what is `Pedigree`?

Comment (danlipsa) [inline Packages/vcs/Lib/vcs2vtk.py]: It is a terminology VTK uses. It is just an index to keep track of where pieces of cells end up after I wrap the dataset around. Certain cells will be cut in two pieces which will end up in different parts of the dataset. When the scalars are updated I use those indexes to set the proper scalar for each cell.

**Comment by danlipsa** [comment]
Sorry, I pushed by mistake my current work on top of this. All tests should pass now.

**Comment by doutriaux1** [comment]
@danlipsa looks like master is too far away from your branch (in baseline) do you mind merging master in?

**Comment by doutriaux1** [comment]
Never mind, it's a bot issue. But got locked out of both bots... Yeah... Will fix tomorrow. Bots seem OK. Will build on my Mac just to be safe (hopefully tonight) and will merge

**Comment by danlipsa** [comment]
@doutriaux1 Sounds good. Thanks Charles.

**Comment by doutriaux1** [comment]
ok passes on my mac as well. Merging!
https://open.cdash.org/testDetails.php?test=412544295&build=4202288

**Pull request merged**